File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "";
    	int c1 = 0;
	int c2 = 0;
	for (auto g : goal)
	{
		if (cards1[c1] == g)
			c1++;
		else if (cards2[c2] == g)
			c2++;
        else break;
	}
	if (c1+c2 == goal.size()) answer = "Yes";
	else answer = "No";
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cs
using System;

public class Solution {
    public string solution(string[] cards1, string[] cards2, string[] goal) {
        string answer = "";
        int c1 = 0;
        int c2 = 0;

        for(int i = 0; i<goal.Length;i++)
        {
            bool chk = false;
            if (c1 < cards1.Length)
            {
                if (goal[i] == cards1[c1])
                {
                    c1++;
                    chk = true;
                }
            }

            if(c2 < cards2.Length)
            {
                if (goal[i] == cards2[c2])
                {
                    c2++;
                    chk = true;
                }
            }

            if(!chk)
            {
                answer = "No";
                break;
            }
        }

        if(!answer.Equals("No"))
        {
            answer = "Yes";
        }
        return answer;
    }
}